pkgs/build-support/compress-drv/default.nix

```diff
@@ -72,7 +68,11 @@ let
 in
 runCommand "${drv.name}-compressed" { } ''
   mkdir $out
-  (cd $out; ${xorg.lndir}/bin/lndir ${drv})
+
+  # cannot use lndir here, because it also symlinks directories,
+  # which we do not need; we only need to symlink files.
+  (cd ${drv}; find -L -type d -exec mkdir -p $out/{} ';')
+  (cd ${drv}; find -L -type f -exec ln -s ${drv}/{} $out/{} ';')
```

Comment:
I also just find this so much easier to reason about than `lndir`. I'm glad to see the return.

Reply:
It is just how lndir functions, that it doesn't recurse into other symlinks. Probably mainly to avoid symlink loops?